assets/js/googlesitekit/notifications/register-defaults.js

```diff
@@ -147,100 +141,91 @@ export function registerDefaults( notificationsAPI ) {
-					startDate,
-					endDate,
-					dimensions: [ 'pagePath' ],
-					metrics: [ { name: 'totalAdRevenue' } ],
-					orderby: [
-						{
-							metric: { metricName: 'totalAdRevenue' },
-							desc: true,
-						},
-					],
-					limit: 3,
-				};
-
-				// Ensure resolution of the report has completed before showing this
-				// notification, since it should only appear when the user has no data in
-				// the report.
-				const report = await resolveSelect(
-					MODULES_ANALYTICS_4
-				).getReport( reportOptions );
-
-				// This notification should only appear when the user has connected their
-				// AdSense and Google Analytics accounts, but has not yet received any data
-				// from linking the accounts. If they have any data from the "linked" report,
-				// we show them a different notification and should not show this one. Check
-				// to see if the user already has data and dismiss this notification without
-				// showing it.
-				if (
-					isZeroReport( report ) === false &&
-					analyticsAndAdsenseConnectedAndLinked
-				) {
-					await dispatch( CORE_NOTIFICATIONS ).dismissNotification(
-						'top-earning-pages-success-notification'
-					);
-					return false;
-				}
+			const adSenseModuleConnected =
+				select( CORE_MODULES ).isModuleConnected( 'adsense' );
 
-				return true;
-			},
-			isDismissible: true,
-		}
-	);
+			const analyticsModuleConnected =
+				select( CORE_MODULES ).isModuleConnected( 'analytics-4' );
 
-	notificationsAPI.registerNotification( 'gathering-data-notification', {
+			const isAdSenseLinked =
+				select( MODULES_ANALYTICS_4 ).getAdSenseLinked();
+
+			const analyticsAndAdsenseConnectedAndLinked =
+				adSenseModuleConnected &&
+				analyticsModuleConnected &&
+				isAdSenseLinked;
+
+			if ( ! analyticsAndAdsenseConnectedAndLinked ) {
+				return false;
+			}
+
+			const { startDate, endDate } = select(
+				CORE_USER
+			).getDateRangeDates( {
+				offsetDays: DATE_RANGE_OFFSET,
+			} );
+
+			const reportOptions = {
+				startDate,
+				endDate,
+				dimensions: [ 'pagePath' ],
+				metrics: [ { name: 'totalAdRevenue' } ],
+				orderby: [
+					{
+						metric: { metricName: 'totalAdRevenue' },
+						desc: true,
+					},
+				],
+				limit: 3,
+			};
+
+			// Ensure resolution of the report has completed before showing this
+			// notification, since it should only appear when the user has no data in
+			// the report.
+			const report = await resolveSelect( MODULES_ANALYTICS_4 ).getReport(
+				reportOptions
+			);
+
+			// This notification should only appear when the user has connected their
+			// AdSense and Google Analytics accounts, but has not yet received any data
+			// from linking the accounts. If they have any data from the "linked" report,
+			// we show them a different notification and should not show this one. Check
+			// to see if the user already has data and dismiss this notification without
+			// showing it.
+			if (
+				isZeroReport( report ) === false &&
```

Comment:
The comment above says this appears when the user has no data, but here the check is that they _do_ have data, since the report is _not a zero report_. Is the comment or the condition here wrong? 🤔 

I know this is from before, but is this comment exactly right? Something here seems a bit off 🤔

Reply:
Both seem to be correct as we are actually "dismissing" the notification and "hiding" it if the report is NOT zero (`isZeroReport` is false), i.e. if there is data.

But yes, this logic and comment were ported over more or less "as is". I don't recollect which "different notification" we are exactly referring to here. But as far as the logic for this notification goes, this seems alright.